checks/s3/check.go

```diff
@@ -0,0 +1,63 @@
+package s3
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// Config is the S3 checker configuration settings container.
+type Config struct {
+	// Region indicates the AWS region. Optional.
+	Region string
+
+	// AccessKeyId is used to create static credentials for an AWS session. Optional.
+	AccessKeyId string
+
+	// SecretAccessKey is used to create static credentials for an AWS session. Optional.
+	SecretAccessKey string
+
+	// SessionToken is used to create static credentials for an AWS session.
+	// The token is for temporary security credentials retrieved via STS. Optional.
+	SessionToken string
+
+	// ListBucketsInput is used as a parameter to retrieve a list of s3 buckets. Optional
+	ListBucketsInput *s3.ListBucketsInput
+}
+
+// New creates new S3 health check that verifies the following:
+// - that a connection can be established to s3 through creating a session
+// - that s3 buckets can be accessed by listing all s3 buckets
+func New(config Config) func(ctx context.Context) error {
+	return func(ctx context.Context) (checkErr error) {
```

Comment:
same situation here, no deferred calls that should change the return, so is easier to no use named returns